Sources/DockProgress/DockProgress.swift

```diff
@@ -250,3 +264,47 @@ public enum DockProgress {
 		}
 	}
 }
+
+private typealias DisplayLinkObserverCallback = (DisplayLinkObserver, Double) -> Void;
+
+private class DisplayLinkObserver {
+    private var displayLink: CVDisplayLink?
+    var callback: DisplayLinkObserverCallback
+    
+    init(_ callback: @escaping DisplayLinkObserverCallback) {
+        self.callback = callback
+        CVDisplayLinkCreateWithActiveCGDisplays(&displayLink)
+    }
+    
+    func start() {
+        if let displayLink {
+            CVDisplayLinkSetOutputCallback(
+                displayLink,
+                displayLinkOutputCallback,
+                UnsafeMutableRawPointer(Unmanaged.passUnretained(self).toOpaque())
+            )
+            CVDisplayLinkStart(displayLink)
+        }
+    }
+    
+    func stop() {
+        if let displayLink {
+            CVDisplayLinkStop(displayLink)
+        }
+    }
+}
+
+private func displayLinkOutputCallback(
+    displayLink: CVDisplayLink,
+    inNow: UnsafePointer<CVTimeStamp>,
+    inOutputTime: UnsafePointer<CVTimeStamp>,
+    flagsIn: CVOptionFlags,
+    flagsOut: UnsafeMutablePointer<CVOptionFlags>,
+    displayLinkContext: UnsafeMutableRawPointer?
+) -> CVReturn {
+    Task { @MainActor in
```

Comment:
This should be in the DockProgress code. This part of the code doesn't have to enforce the mainactor.